### PR TITLE
Add "SHOW CLUSTER(s)" queries support

### PR DESCRIPTION
**Pull request by hexiaoting**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature
 

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add SHOW CLUSTER(S) queries

Detailed description / Documentation draft:

```
SHOW CLUSTERS;
SHOW CLUSTERS LIKE 'xxx';
SHOW CLUSTER 'xxx';
```

reference to issue #11411

### CODE REVIEW

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include <Common/typeid_cast.h>
 #include <iomanip>
 #include <sstream>
+#include <iostream>
```
Comment (tavplubix) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:11]: This include looks redundant

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 2]
```diff
@@ -33,6 +34,32 @@ String InterpreterShowTablesQuery::getRewrittenQuery()
     if (query.databases)
         return "SELECT name FROM system.databases";
 
+    /// SHOW CLUSTER/CLUSTERS 
+    if (query.clusters)
+    {
+        std::stringstream rewritten_query;
+	rewritten_query << "SELECT cluster FROM system.clusters";
```
Comment (tavplubix) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:41]: Do you mean `SELECT * FROM ...`?

**Review by tavplubix — COMMENTED** [review]
Please fix errors found by Style check.

File: src/Interpreters/InterpreterShowTablesQuery.cpp (hunk 2 above)
Comment (hexiaoting) [reply src/Interpreters/InterpreterShowTablesQuery.cpp:41]: No, I mean just show cluster name while using "show clusters", 
eg:
```
dell124 :) show clusters;

SHOW CLUSTERS

┌─cluster───────────────────────────┐
│ test_cluster_two_shards           │
│ test_cluster_two_shards           │
│ test_cluster_two_shards_localhost │
│ test_cluster_two_shards_localhost │
│ test_shard_localhost              │
│ test_shard_localhost[1]           │
│ test_shard_localhost_secure       │
│ test_unavailable_shard            │
│ test_unavailable_shard            │
└───────────────────────────────────┘

9 rows in set. Elapsed: 0.002 sec.
```
Do I need to show all the info about clusters, like this:
```
┌─cluster───────────────────────────┬─shard_num─┬─shard_weight─┬─replica_num─┬─host_name─┬─host_address─┬─port─┬─is_local─┬─user────┬─default_database─┬─errors_count─┬─estimated_recovery_time─┐
│ test_cluster_two_shards           │         1 │            1 │           1 │ 127.0.0.1 │ 127.0.0.1    │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_cluster_two_shards           │         2 │            1 │           1 │ 127.0.0.2 │ 127.0.0.2    │ 9000 │        0 │ default │                  │            0 │                       0 │
│ test_cluster_two_shards_localhost │         1 │            1 │           1 │ localhost │ ::1          │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_cluster_two_shards_localhost │         2 │            1 │           1 │ localhost │ ::1          │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_shard_localhost              │         1 │            1 │           1 │ localhost │ ::1          │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_shard_localhost[1]           │         1 │            1 │           1 │ localhost │ ::1          │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_shard_localhost_secure       │         1 │            1 │           1 │ localhost │ ::1          │ 9440 │        0 │ default │                  │            0 │                       0 │
│ test_unavailable_shard            │         1 │            1 │           1 │ localhost │ ::1          │ 9000 │        1 │ default │                  │            0 │                       0 │
│ test_unavailable_shard            │         2 │            1 │           1 │ localhost │ ::1          │    1 │        0 │ default │                  │            0 │                       0 │
└───────────────────────────────────┴───────────┴──────────────┴─────────────┴───────────┴──────────────┴──────┴──────────┴─────────┴──────────────────┴──────────────┴─────────────────────────┘

```

@tavplubix  @alexey-milovidov

Comment (tavplubix) [reply src/Interpreters/InterpreterShowTablesQuery.cpp]: Ok, let `SHOW CLUSTERS` show just names and `SHOW CLUSTER name` show all the info about the cluster. But then it should be `SELECT DISTINCT cluster FROM ...` to show each name only once in `SHOW CLUSTERS`.

Comment (hexiaoting) [reply src/Interpreters/InterpreterShowTablesQuery.cpp]: OK, I will fix this by adding "DISTINCT".

**Pull request merged**